src/Model/DocsAsHtml.hs

```diff
@@ -203,10 +203,18 @@ linkToSource (LinksContext{..}, _) (P.SourceSpan name start end) =
   (P.SourcePos endLine _) = end
   (GithubUser user, GithubRepo repo) = ctxGithub
 
-  relativeToBase = intercalate "/" . dropWhile (/= "src") . splitOn "/"
+  relativeToBase = intercalate "/" . dropWhile (/= "src") . splitOnPathSep
   githubBaseUrl = concat ["https://github.com/", user, "/", repo]
   fragment = "L" ++ show startLine ++ "-L" ++ show endLine
 
+-- | Split a string on either unix-style "/" or Windows-style "\\" path
+-- | separators.
+splitOnPathSep :: String -> [String]
```

Comment:
I think regexes don't belong in real code; in most cases they're near write-only and very difficult to debug. The one you describe would be fairly straightforward, but I can't see an improvement over what's currently here.

Also, I don't know how to split on a regex in Haskell, because of the above :)